easycv/toolkit/quantize/quantize_utils.py

```diff
@@ -1,19 +1,179 @@
 # Copyright (c) Alibaba, Inc. and its affiliates.
+import itertools
+
+import mmcv
+import MNN
+import numpy as np
 import torch
 from mmcv.parallel import scatter_kwargs
 from mmcv.runner import get_dist_info
 
-quantize_config = {
-    'device': 'cpu',
-    'backend': 'PyTorch',
-}
+from easycv.models.detection.utils import postprocess, yolox_predict
+from easycv.models.detection.yolox.yolo_head import YOLOXHead
+
+
+def quantize_config_check(device, backend, model_type=''):
+    """
+    check quantize config and return config setting
+    """
+    quantize_config = {}
+    if device == 'cpu' and backend == 'PyTorch':
+        quantize_config['device'] = 'cpu'
+        quantize_config['backend'] = 'PyTorch'
+    elif device == 'arm' and backend == 'MNN':
+        quantize_config['device'] = 'arm'
+        quantize_config['backend'] = 'MNN'
+    else:
+        raise ValueError(
+            '{} device and {} backend is not supported. It can only be (cpu, PyTorch) or (arm, MNN).'
+            .format(device, backend))
+
+    if 'YOLOX' in model_type:
+        quantize_config['non_traceable_module_class'] = [YOLOXHead]
+
+    return quantize_config
 
 
 def calib(model, data_loader):
     for cur_iter, data in enumerate(data_loader):
-        input_args, kwargs = scatter_kwargs(None, data, [-1])
+        if cur_iter > 50:
+            return
+        input_args, kwargs = scatter_kwargs(None, data, [torch.cuda.current_device()]) #[-1]
+        print('calib number: {}'.format((cur_iter)))
         with torch.no_grad():
-            kwargs[0]['img'] = kwargs[0]['img'].squeeze(dim=0)
+            # for cpu test,  use this:
+            # kwargs[0]['img'] = kwargs[0]['img'].squeeze(dim=0)
             model(kwargs[0]['img'])
```

Comment:
cpu gpu test logic may use if condition to make better compatibility

Reply:
If calib use cpu, it will cost too much time.